mlir/lib/Dialect/Tensor/IR/TensorOps.cpp

```diff
@@ -1982,14 +1982,74 @@ struct FoldDimOfCollapseShape : public OpRewritePattern<DimOp> {
     return success();
   }
 };
+
+struct ConvertToStaticExpandShape : public OpRewritePattern<ExpandShapeOp> {
+  using OpRewritePattern<ExpandShapeOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(ExpandShapeOp expandOp,
+                                PatternRewriter &rewriter) const override {
+    SmallVector<int64_t> newOutputShape(expandOp.getResultType().getShape());
```

Comment:
You should look for source of `expandOp` is a `tensor.cast` operation where the source of the cast has a more static shape than the result (using https://github.com/llvm/llvm-project/blob/fb858b4f6ba454565abecbc107364a444668840b/mlir/include/mlir/Dialect/Tensor/IR/Tensor.h#L88 ).